target/linux/ath79/image/generic.mk

```diff
@@ -13,30 +13,36 @@ define Device/avm_fritz300e
 endef
 TARGET_DEVICES += avm_fritz300e
 
+define Device/buffalo_wzr-hp-g450h
+  ATH_SOC := ar7242
+  DEVICE_TITLE := Buffalo WZR-HP-G450H
+  DEVICE_PACKAGES := kmod-usb-core kmod-usb2 kmod-usb-ledtrig-usbport
+  IMAGE_SIZE := 32256k
+  SUPPORTED_DEVICES += wzr-hp-g450h
+endef
+TARGET_DEVICES += buffalo_wzr-hp-g450h
+
 define Device/embeddedwireless_dorin
   ATH_SOC := ar9331
   DEVICE_TITLE := Embedded Wireless Dorin
   DEVICE_PACKAGES := kmod-usb-chipidea2
   IMAGE_SIZE := 16000k
-  SUPPORTED_DEVICES += ew-dorin
 endef
 TARGET_DEVICES += embeddedwireless_dorin
 
-define Device/etactica-eg200
+define Device/etactica_eg200
   ATH_SOC := ar9331
   DEVICE_TITLE := eTactica EG200
   DEVICE_PACKAGES := kmod-usb-chipidea2 kmod-ledtrig-oneshot \
 	kmod-usb-serial kmod-usb-serial-ftdi kmod-usb-storage  kmod-fs-ext4
-  SUPPORTED_DEVICES += etactica,eg200 rme-eg200
```

Comment:
why are you dropping these? this is how you can do upgrades from older images that used older names, like from ar71xx to ath79 and so on.  (You're dropping them all over the place)  without these, you get image validation failures on sysupgrade.

Reply:
(if anything, you should be _adding_ things here, because you're renaming everything!)

Reply:
The idea was that changing from ar71xx to ath79 should require a forced sysupgrade, as config has changed at several places.  I'm not a fan of this policy, because I use an OpenWRT derivative which does not copy config anyway, but this is the reason why supported_devices are removed.

Reply:
its a philosophical question. I am still hoping to find time to get qca8k working on the builtin switches. at that point sysupgrade would break anyhow.

Reply:
I would suggest that dropping all the supported devices should be an unrelated commit.  it's rather hidden at the moment inside "cleanup".

Remember, you're also breaking updates from ath79 to ath79 by your renaming here, even though it's still within master codebase.

Reply:
Similar things was done by @mkreslin when he made some board rename in ramips so I dropped them instead of adding SUPPORTED_DEVICES to make sysupgrade work.
And I said about this in commit description (because title is too long)
I doubt whether it's needed to drop one line using two separated commits. (1 Drop the <manufacturer>,<board> because it's automatically generated. 2 drop the one coming from ar71xx because we shouldn't allow a sysupgrade from ar71xx to ath79 with config files saved.)

Reply:
BTW the only broken sysupgrade from ath79 to ath79 here is Archer C7 because I changed the board name.

Reply:
PS: I'm not renaming everything. the dts/image renaming is only about filename, and what sysupgrade check is the first compatible string in dts. I changed the compatible string for Archer C7 and WR740N/WR741N v4. I said about the Archer C7 renaming in a separated commit. As for WR740N/WR741N, what the previous author submitted is completely broken so I don't think anyone has been using that image successfully.(board name in scripts doesn't match dts compatible so the default configuration isn't even generated.)

Reply:
Since it caused an argument here, I'll make an RFC patch for dropping compatible strings from ar71xx on mailing list.